modules/core/core-api/src/main/java/com/enonic/xp/app/ApplicationWildcardMatcher.java

```diff
@@ -30,17 +26,16 @@ public boolean matches( final String wildcard, T input )
     public Predicate<T> createPredicate( final String wildcard )
     {
         final String resolvedAppName = resolveAppPlaceholder( wildcard, applicationKey );
-        final Predicate<String> predicate = resolvedAppName.contains( ANY_WILDCARD )
-            ? Pattern.compile( resolvedAppName.replace( "*", ".*" ) ).asMatchPredicate()
-            : resolvedAppName::equals;
+
+        final Predicate<String> predicate =
+            Pattern.compile( resolvedAppName.replace( "*", ".*" ) ).
+                asMatchPredicate();
 
         return t -> predicate.test( toNameFunction.apply( t ) );
     }
 
     private static String resolveAppPlaceholder( final String nameToResolve, final ApplicationKey applicationKey )
     {
-        return nameToResolve.startsWith( APP_WILDCARD_PREFIX )
-            ? applicationKey + nameToResolve.substring( APP_WILDCARD.length() )
-            : nameToResolve;
+        return nameToResolve.replace( APP_WILDCARD, applicationKey.getName().replace( ".", "\\." ) );
```

Comment:
it may create more issues. Use `Matcher.quoteReplacement` together with `Pattern.quote`